pkg/piperutils/fileUtils_test.go

```diff
@@ -0,0 +1,43 @@
+package piperutils
+
+import (
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestFileExists(t *testing.T) {
+	dir, err := ioutil.TempDir("", "dir")
```

Comment:
I would like to suggest to introduce some layer where we can mock file system access rather than dealing with temp directories.

* We don*t have to consider problems with creating the tmp dir
* When working with tmp directories the code under tests needs to be prepared so that it can work with tmp directories. Sometimes that is not handy, e.g. when we place files in some certain fixed directories (e.g. home directory). In that cases it makes the code unnecessarily complex.
* We can directly in the mock check if a certain file related action (e.g. read) has been performed by setting a flag from the mock and check that after test.

Reply:
In my opinion it is ok to go with tmp file system as also indicated by https://github.com/SAP/jenkins-library/issues/1016#issuecomment-559688250. 

This is what golang standard libraries do as well and experiences so far have been purely positive.